### ISSUE TITLE
Confusing staticcheck rule activation 

### ISSUE THREAD

**Issue by marten-seemann** [golangci/golangci-lint#1649]
<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.35.2 built from 1da5701 on 2021-01-11T02:54:03Z
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
no config
```

</details>

<details><summary>Go environment</summary>

```console
$ go version 
go version go1.15.6 darwin/amd64
```

</details>

Consider the following minimal example:
```go
package main

import "fmt"

type user struct {
	name string
	age  int
}

func main() {
	fmt.Println(getHello())

	u := &user{name: "John Doe"}
	fmt.Println(u)
}

func getHello() string {
	return fmt.Sprintf("Hallo")
}
```

Running `staticcheck .` finds the following problems:
```
main.go:7:2: field age is unused (U1000)
main.go:18:9: unnecessary use of fmt.Sprintf (S1039)
```
Running `golangci-lint run --disable-all --enable staticcheck -v.` doesn't output any issues:
```
INFO [config_reader] Config search paths: XXX
INFO [lintersdb] Active 1 linters: [staticcheck]
INFO [loader] Go packages loading at mode 575 (compiled_files|files|imports|name|types_sizes|deps|exports_file) took 214.13809ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 168.483µs
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [runner] processing took 5.779µs with stages: max_same_issues: 1.726µs, nolint: 1.454µs, filename_unadjuster: 418ns, skip_dirs: 259ns, cgo: 184ns, skip_files: 180ns, identifier_marker: 175ns, max_from_linter: 171ns, severity-rules: 125ns, exclude: 123ns, autogenerated_exclude: 120ns, path_prefixer: 118ns, uniq_by_line: 117ns, source_code: 117ns, path_prettifier: 114ns, diff: 112ns, path_shortener: 108ns, max_per_file_from_linter: 55ns, sort_results: 54ns, exclude-rules: 49ns
INFO [runner] linters took 21.253215ms with stages: staticcheck: 21.097502ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 4 samples, avg is 73.2MB, max is 73.6MB
INFO Execution took 252.932387ms
```

It exits with status code 0, i.e. it doesn't find any issues with the code.

**Comment by ldez** [comment]
```
$ golangci-lint run
cmd/foo.go:7:2: `age` is unused (structcheck)
        age  int
        ^
cmd/foo.go:18:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
        return fmt.Sprintf("Hallo")
               ^
```

staticcheck is in fact splitter in different parts: `gosimple`, `structcheck`

**Comment by marten-seemann** [comment]
Should it be removed from the list of linters then, or replaced by an error / warning telling the user exactly which linters it was split up into? Running `golangci-lint run --disable-all --enable staticcheck` and not getting any errors is pretty confusing in my opinion.

**Comment by ldez** [comment]
More explanation about staticcheck configuration in: https://github.com/golangci/golangci-lint/discussions/1989#discussioncomment-745004

Closed in favor of #357
